include/nbl/builtin/hlsl/complex.hlsl

```diff
@@ -12,178 +12,178 @@ namespace nbl
+        result.m_imag = (m_imag * rhs.m_real - m_real * rhs.m_imag) / denominator;
 
-            return result;
-        }
-            
-        // ----------------- Relational operators -----------------------------
-        bool operator==(const complex_t rhs)
-        {
-                return !(uint64_t(m_real) ^ uint64_t(rhs.m_real)) && !(uint64_t(m_imag) ^ uint64_t(rhs.m_imag));
-        }
-        bool operator!=(const complex_t rhs)
-        {
-                return uint64_t(m_real) ^ uint64_t(rhs.m_real) || uint64_t(m_imag) ^ uint64_t(rhs.m_imag);
-        }
+        return result;
+    }
             
-    };
-
-    // ---------------------- Compound assignment functors ----------------------
-    template <typename Scalar>
-    struct ComplexCompoundHelper{              
-        using add = assign_add_t<complex_t<Scalar> >;
-        using subtract = assign_subtract_t<complex_t<Scalar> >;
-        using mul = assign_mul_t<complex_t<Scalar> >;
-        using div = assign_div_t<complex_t<Scalar> >;
-    };
-
-    template <typename Scalar>
-    using assign_add_complex = typename ComplexCompoundHelper<Scalar>::add;
-    template <typename Scalar>
-    using assign_subtract_complex = typename ComplexCompoundHelper<Scalar>::subtract;
-    template <typename Scalar>
-    using assign_mul_complex = typename ComplexCompoundHelper<Scalar>::mul;
-    template <typename Scalar>
-    using assign_div_complex = typename ComplexCompoundHelper<Scalar>::div;
-
-    // ---------------------- Non-member functions -----------------------------    
-    template<typename Scalar>
-    Scalar real(const complex_t<Scalar> c) {
-        return c.m_real;
+    // ----------------- Relational operators -----------------------------
+    bool operator==(const complex_t rhs)
+    {
+            return !(uint64_t(m_real) ^ uint64_t(rhs.m_real)) && !(uint64_t(m_imag) ^ uint64_t(rhs.m_imag));
     }
-    
-    template<typename Scalar>
-    Scalar imag(const complex_t<Scalar> c) {
-        return c.m_imag;
+    bool operator!=(const complex_t rhs)
+    {
+            return uint64_t(m_real) ^ uint64_t(rhs.m_real) || uint64_t(m_imag) ^ uint64_t(rhs.m_imag);
     }
+            
+};
+
+// ---------------------- Compound assignment functors ----------------------
+template <typename Scalar>
+struct ComplexCompoundHelper{              
+    using add = assign_add_t<complex_t<Scalar> >;
+    using subtract = assign_subtract_t<complex_t<Scalar> >;
+    using mul = assign_mul_t<complex_t<Scalar> >;
+    using div = assign_div_t<complex_t<Scalar> >;
+};
+
+template <typename Scalar>
+using assign_add_complex = typename ComplexCompoundHelper<Scalar>::add;
+template <typename Scalar>
+using assign_subtract_complex = typename ComplexCompoundHelper<Scalar>::subtract;
+template <typename Scalar>
+using assign_mul_complex = typename ComplexCompoundHelper<Scalar>::mul;
+template <typename Scalar>
+using assign_div_complex = typename ComplexCompoundHelper<Scalar>::div;
```

Comment:
if you need aliases, then don't have them hanging around globally imho, or we'll drown